### ISSUE TITLE
Appium fails node security check (nsp) for appium-fake-driver dependency

### ISSUE THREAD

**Issue by jamie-sherriff** [appium/appium#8877]
## The problem
Hello,  I happend to run nsp [https://www.npmjs.com/package/nsp](https://www.npmjs.com/package/nsp) in my project and noticed Appium fails for [https://nodesecurity.io/advisories/118](https://nodesecurity.io/advisories/118)  under the appium-fake-driver module.

Will you accept a PR that bumps babel-core to the latest which includes minimatch@3.0.2
(Posting here because appium-fake-driver issue tracker is disabled).

I don't know the full impact of bumping babel-core in  appium-fake-driver, will that cause any issues that you can think of?

## Environment

* Assuming all versions (Verified on 1.6 beta release branch) (24efa6edf0eabd87e21e2e7f940e39192e89dc03)
* Node.js V6.10

## Details
$ nsp check
(+) 1 vulnerabilities found
┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────┐
│               │ Regular Expression Denial of Service                                                          │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ Name          │ minimatch                                                                                     │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ CVSS          │ 7.5 (High)                                                                                    │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ Installed     │ 2.0.10                                                                                        │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ Vulnerable    │ <=3.0.1                                                                                       │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ Patched       │ >=3.0.2                                                                                       │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ Path          │ appium@1.6.6-beta.3 > appium-fake-driver@0.1.11 > babel-core@5.8.24 > minimatch@2.0.10        │
├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/118                                                        │
└───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────┘

**Comment by dpgraham** [comment]
I don't think it will cause any issues. You can try bumping it, as long as the tests still pass it should be okay.

**Comment by jamie-sherriff** [comment]
Unfortunately i went from 4 unit and 111 functional passing tests down to only 2 passing units tests and 0 passing functional tests after  updating the babel dependencies. Also it seems appium-gulp-plugins has the same security issue with minimatch as a  dev-dependency.
The main repeating error was 
` TypeError: Cannot read property 'async' of undefined`

I will continue to look into this and see how far i get.
I also noticed one test was failing on the head of appium-fake-driver for me prior to the upgrade

**Comment by imurchie** [comment]
Babel cannot be updated without significant work on the plumbing that builds Appium, unfortunately. We haven't had time and that is why it hasn't been done.

**Comment by jamie-sherriff** [comment]
Right, i have just removed babel-core and babel-runtime from the dependencies in fake-driver and the tests pass after a full clear of the node)modules directory (there is 1 test failing that was already and ill look at that)

From what i can tell is that babel should get inherited from appium-gulp-plugins(which i haven't bumped the version of babel on)? 

Assuming i can fix this failing test can anyone for see any other problems?

**Comment by jamie-sherriff** [comment]
Ok i have found the problem with the failing test.
It passes in appium-base-driver@2.3.0 (version used last time it passed on travis) but fails on the latest 2.11.0

Can i get away with fixing the base-driver version to 2.3.0?

Test failing is:
~~~
  1) BaseDriver event timing framework "before each" hook: callee$3$0 for "should allow logging the same event multiple times":
     TypeError: path must be a string or Buffer
      at TypeError (native)
~~~

This can't be representing an actual problem in appium can it? as that test will be failing

**Comment by imurchie** [comment]
I don't understand what you've done and what you are trying to do, anymore.

**Comment by jamie-sherriff** [comment]
@imurchie Yea sorry, was abit unclear there. The goal is still to get Appium to pass a nsp check for production dependencies.

1. I have removed babel-core and babel-runtime from appium-fake-driver dependencies
2. I have fixed the appium-base-driver version to 2.3.0 so the e2e tests pass

I will do a PR and you can see it.

**Comment by jamie-sherriff** [comment]
Closing as completed
